lib/rspec/support/differ.rb

```diff
@@ -95,18 +96,18 @@ def diffably_stringify(array)
           if Array === entry
             entry.inspect
           else
-            entry.to_s.gsub("\n", "\\n")
+            entry.to_s.gsub(LINE_BREAK, Regexp.escape(LINE_BREAK))
```

Comment:
So the pre-escaped `"\\n"` still seems like a better idea than running an escape each time this is used.